lib/App/DuckPAN/Perl.pm

```diff
@@ -109,54 +111,73 @@ sub duckpan_install {
 		$sp =~ s/\:\:/_/g;
 
 		# special case: check for a pinned verison number
-		my $pin_version            = $ENV{$sp};
-		my $localver               = $self->get_local_version($package);
-		my $duckpan_module_version = version->parse($module->version);
-		my $duckpan_module_url     = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
+		my $pinned_version      = $ENV{$sp};
+		my $installed_version   = $self->get_local_version($package);
+		my $latest_version      = version->parse($module->version);
+		my $duckpan_module_url  = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
 
-		$localver ||= 1e-6 if ($pin_version); # a silly, but true, value if missing and we need to compare with pinned.
```

Comment:
This line actually doesn't prevent that.  When I reconsidered what _reinstall_ should do I wanted to limit it to literally reinstalling the installed version.  It doesn't quite get there because the same if block handles the "not installed" case as well but it at least defaults to it (line 125.)  So you're trying to reinstall a version that's essentially not installed.  That is a fatal error imo.  How could it possibly reinstall that?

If you replace DuckPAN.pm and Perl.pm (or install it), you should see the desired behavior.  We could also check for 9.999 in this block and skip it.  Except for updating the reinstall logic and needing to test, are we really going to be doing a lot of reinstalling with local builds?